### DISCUSSION TITLE
While User onBoarding, Location search is giving empty response (States list)

### DISCUSSION THREAD

**Discussion by vinukumar-vs** [Sunbird-Lern/Community#92] [Issues] [answered]
We have noticed this issue while setting-up multiple infra's. Documentation of the reason for failure will be useful for any adopter. Please share the documentation link if it is already exist. 

> curl --location -g '{{host}}/learner/data/v1/location/search' \
--header 'Content-Type: application/json' \
--header 'Authorization: {{kong_api_key}}' \
--data '{
    "params": {},
    "request": {
        "filters": {
            "type": "state"
        }
    }
}'

Response:

> {
    "id": "api.location.search",
    "ver": "v1",
    "ts": "2023-03-22 17:31:06:723+0000",
    "params": {
        "resmsgid": "b5e67acf78f706a12a22bb404d69d8bc",
        "msgid": "b5e67acf78f706a12a22bb404d69d8bc",
        "err": null,
        "status": "SUCCESS",
        "errmsg": null
    },
    "responseCode": "OK",
    "result": {
        "response": [],
        "count": 0
    }
}

@reshmi-nair @rhwarrier @krishnaj20 @gandham-santhosh @rajeevsathish

**Comment by reshmi-nair** [comment]
@vinukumar-vs Pls check if the cassandra db sunbird keyspace location table has data or not. Also pls if the location Elastic search index is setup properly. If data is not there, pls create data using location create API or by using /v1/bulk/location/upload API.

**Reply by reshmi-nair** [reply]
@vinukumar-vs Upload api details 
curl --location 'https://dev.lern.sunbird.org//api/data/v1/bulk/location/upload' \
--header '' \
--header 'x-authenticated-user-token: ' \
--form 'file=@"/D:/Test data/API upload files/location_upload (1).csv"' \
--form 'type="school"'

csv file data sample:
code,name,parentCode,parentId
skool-01,skool-01,clust-01,d4e62d73-948b-4436-9061-abdb6ae54c1f
skool-02,skool-02,clust-01,d4e62d73-948b-4436-9061-abdb6ae54c1f

**Reply by nidhikachhawa24** [reply]
@reshmi-nair 
From where can I get the CSV data? Or if we've to create then how because the sample data you've mentioned is having the codes?

**Reply by vinukumar-vs** [reply]
@reshmi-nair Can we use the same location API to upload school & blocks information as well.

https://lern.sunbird.org/learn/product-and-developer-guide/user-and-org-service/api-documentation/backend-services

**Reply by reshmi-nair** [reply, marked as answer]
yes, just change the type form field and pass the relevant csv with parent id as district id.
--form 'type="block"'
